src/main/groovy/org/pih/warehouse/inventory/CycleCountSubmitCountCommand.groovy

```diff
@@ -0,0 +1,39 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.Validateable
+import org.pih.warehouse.core.Location
+import org.springframework.web.context.request.RequestContextHolder
+
+class CycleCountSubmitCountCommand implements Validateable {
+
+    CycleCount cycleCount
+
+    Location facility
+
+    boolean refreshQuantityOnHand
+
+    boolean failOnOutdatedQuantity
+
+    boolean requireRecountOnDiscrepancy
+
+    def beforeValidate() {
+        String locationId = RequestContextHolder.getRequestAttributes().params?.facility
+        String cycleCountId = RequestContextHolder.getRequestAttributes().params?.cycleCountId
+        facility = Location.read(locationId)
+        cycleCount = CycleCount.read(cycleCountId)
+    }
+
+    static constraints = {
+        cycleCount(validator: { CycleCount obj ->
+            List<CycleCountStatus> validStatuses = [
+                    CycleCountStatus.COUNTING,
+                    CycleCountStatus.INVESTIGATING,
+                    CycleCountStatus.REQUESTED,
+                    CycleCountStatus.COUNTED,
```

Comment:
quote from docs:

> If Cycle Count is not in status ready to count, counting, investigating, or counted, error

1. is `READY_TO_COUNT` equivalent to `REQUESTED`?
2. should the `COUNTED` be there? If so, I can spam the requests for the same cycle count and eventually create X transactions for the same cycle count, not really mutating anything (duplicates would be created)

Reply:
I wrote those statuses before I really knew which ones we were using. Given that if you've started the count you'll be in status COUNTING, and given that we're going to split count and recount into their own APIs, I think we only need COUNTING here actually.